diff --git a/tpde-llvm/src/encode_template.c b/tpde-llvm/src/encode_template.c
index b35fd9b..29f94bf 100644
--- a/tpde-llvm/src/encode_template.c
+++ b/tpde-llvm/src/encode_template.c
@@ -727,5 +727,9 @@ void TARGET_V1 fence_acq(void) { __atomic_thread_fence(__ATOMIC_ACQUIRE); }
 void TARGET_V1 fence_rel(void) { __atomic_thread_fence(__ATOMIC_RELEASE); }
 void TARGET_V1 fence_acqrel(void) { __atomic_thread_fence(__ATOMIC_ACQ_REL); }
+#ifndef __x86_64__
 void TARGET_V1 fence_seqcst(void) { __atomic_thread_fence(__ATOMIC_SEQ_CST); }
+#else
+void TARGET_V1 fence_seqcst(void) { __builtin_ia32_mfence(); }
+#endif
 
 // --------------------------
